tests/mock_tests/portsorch_ut.cpp

```diff
@@ -726,32 +648,37 @@ namespace portsorch_test
         static_cast<Orch *>(gBufferOrch)->doTask();
 
         ASSERT_TRUE(current_create_buffer_pool_count == _sai_create_buffer_pool_count);
-        ASSERT_TRUE(gBufferOrch->m_ingressZeroPoolRefCount == 1);
-        ASSERT_TRUE(gBufferOrch->m_egressZeroPoolRefCount == 0);
-        ASSERT_TRUE(gBufferOrch->m_ingressZeroBufferPool != SAI_NULL_OBJECT_ID);
-        ASSERT_TRUE(gBufferOrch->m_egressZeroBufferPool == SAI_NULL_OBJECT_ID);
+        ASSERT_TRUE(gBufferOrch->m_ingressZeroPoolRefCount == 0);
+        ASSERT_TRUE(gBufferOrch->m_egressZeroPoolRefCount == 1);
+        ASSERT_TRUE(gBufferOrch->m_ingressZeroBufferPool == SAI_NULL_OBJECT_ID);
+        ASSERT_TRUE(gBufferOrch->m_egressZeroBufferPool != SAI_NULL_OBJECT_ID);
 
         auto countersTable = make_shared<Table>(m_counters_db.get(), COUNTERS_TABLE);
         auto dropHandler = make_unique<PfcWdZeroBufferHandler>(port.m_port_id, port.m_queue_ids[3], 3, countersTable);
 
-        current_create_buffer_pool_count++; // Increased for egress zero pool
         ASSERT_TRUE(current_create_buffer_pool_count == _sai_create_buffer_pool_count);
-        ASSERT_TRUE(PfcWdZeroBufferHandler::ZeroBufferProfile::getInstance().getPool(true) == gBufferOrch->m_ingressZeroBufferPool);
-        ASSERT_TRUE(PfcWdZeroBufferHandler::ZeroBufferProfile::getInstance().getPool(false) == gBufferOrch->m_egressZeroBufferPool);
-        ASSERT_TRUE(gBufferOrch->m_ingressZeroPoolRefCount == 2);
-        ASSERT_TRUE(gBufferOrch->m_egressZeroPoolRefCount == 1);
+        ASSERT_TRUE(PfcWdZeroBufferHandler::ZeroBufferProfile::getInstance().getPool() == gBufferOrch->m_egressZeroBufferPool);
+        ASSERT_TRUE(gBufferOrch->m_ingressZeroPoolRefCount == 0);
+        ASSERT_TRUE(gBufferOrch->m_egressZeroPoolRefCount == 2);
 
         std::deque<KeyOpFieldsValuesTuple> entries;
-        entries.push_back({"ingress_zero_pool", "DEL", {}});
+        entries.push_back({"egress_zero_pool", "DEL", {}});
         poolConsumer->addToSync(entries);
         entries.clear();
         auto current_remove_buffer_pool_count = _sai_remove_buffer_pool_count;
         static_cast<Orch *>(gBufferOrch)->doTask();
-        ASSERT_TRUE(gBufferOrch->m_ingressZeroPoolRefCount == 1);
+        ASSERT_TRUE(gBufferOrch->m_egressZeroPoolRefCount == 1);
         ASSERT_TRUE(_sai_remove_buffer_pool_count == current_remove_buffer_pool_count);
 
         // release zero buffer drop handler
         dropHandler.reset();
+
+        // Destory the Singleton Class
+        PfcWdZeroBufferHandler::ZeroBufferProfile::getInstance().destroyZeroBufferProfile();
+
+        // Check the ref count
+        ASSERT_TRUE(gBufferOrch->m_egressZeroPoolRefCount == 0);
```

Comment:
Should we check whether the egress zero buffer pool has been removed and whether m_egressZeroBufferPool is NULL in this case?

Reply:
Updated, please check